### ISSUE TITLE
kcli delete vm tries to delete storage pool directory

### ISSUE THREAD

**Issue by wzzrd** [karmab/kcli#179]
I use libvirt and have multiple storage pools: "first", "second", "default" and "vm-base-images". All of my images are in the vm-base-images pool. The default pool for kcli is "first".

I start new vms with

`kcli create vm someName -p rhel7`

which works fine. I get an image in pool "first" with the rhel7 image in vm-base-images as the backing store.

Removing that vm gives me the following error, however. It seems like kcli tries to remove the whole directory "first" is mounted at:
```
kcli delete vm someName
Are you sure? [y/N]: y
Deleting vm devnation on local
Traceback (most recent call last):
  File "/usr/bin/kcli", line 11, in <module>
    load_entry_point('kcli==99.0', 'console_scripts', 'kcli')()
  File "/usr/lib/python3.7/site-packages/kvirt/cli.py", line 2545, in cli
    args.func(args)
  File "/usr/lib/python3.7/site-packages/kvirt/cli.py", line 180, in delete_vm
    result = k.delete(name, snapshots=snapshots)
  File "/usr/lib/python3.7/site-packages/kvirt/kvm/__init__.py", line 1587, in delete
    volume.delete(0)
  File "/usr/lib64/python3.7/site-packages/libvirt.py", line 3593, in delete
    if ret == -1: raise libvirtError ('virStorageVolDelete() failed', vol=self)
libvirt.libvirtError: cannot remove directory '/var/lib/libvirt/images/first': Device or resource busy
```

**Comment by karmab** [comment]
i tried to replicate your issue, but deletion worked fine in my case :(
can you redo those steps, but maybe attach the output of `kcli --debug info vm someName` in the middle?

**Comment by wzzrd** [comment]
Sure, I debugged every step :)

```
➜ kcli --debug create vm karim -p rhel8reg
Deploying vm karim from profile rhel8reg...
Rounding up first disk to 20Gb
<domain type='kvm' >
                  <name>karim</name>
                  <metadata>
        <kvirt:info xmlns:kvirt="kvirt">
        <kvirt:creationdate>18-02-2020 08:14</kvirt:creationdate>
        <kvirt:profile>rhel8reg</kvirt:profile>
                        <kvirt:image>rhel-8.1-x86_64-kvm.qcow2</kvirt:image>
                    <kvirt:plan>kvirt</kvirt:plan>
                    </kvirt:info>
                    </metadata>
                  
                  
                  
                  <memory unit='MiB'>2048</memory>
                  <vcpu>2</vcpu>
                  <os>
                    <type arch='x86_64' machine='pc'>hvm</type>
                    
                    <boot dev='hd'/>
                    
                    <bootmenu enable='yes'/>
                  </os>
                  <features>
                    <acpi/>
                    <apic/>
                    <pae/>
                  </features>
                  <clock offset='utc'/>
                  <on_poweroff>destroy</on_poweroff>
                  <on_reboot>restart</on_reboot>
                  <on_crash>restart</on_crash>
                  <devices>
                    <disk type='file' device='disk'>
                    <driver name='qemu' type='qcow2'/>
                    <source file='/var/lib/libvirt/images/first/karim_0.img'/>
                    <backingStore type='file' index='1'>
                                        <format type='qcow2'/>
                                        <source file='/var/lib/libvirt/images/vm-base-images/rhel-8.1-x86_64-kvm.qcow2'/>
                                        </backingStore>
                    <target dev='vda' bus='virtio'/>
                    
                    </disk>
                    
                     <interface type='network'>
                     
                     <source network='default'/>
                     
                     <model type='virtio'/>
                     </interface>
                    <disk type='file' device='cdrom'>
                        <driver name='qemu' type='raw'/>
                        <source file='/var/lib/libvirt/images/first/karim.ISO'/>
                        <target dev='hdd' bus='ide'/>
                        <readonly/>
                        </disk>
                    <input type='tablet' bus='usb'/>
                        <input type='mouse' bus='ps2'/>
                        <graphics type='spice' port='-1' autoport='yes' listen='127.0.0.1'>
                        <listen type='address' address='127.0.0.1'/>
                        </graphics>
                        <memballoon model='virtio'/>
                    <serial type='pty'>
                       <target port='0'/>
                       </serial>
                       <console type='pty'>
                       <target type='serial' port='0'/>
                       </console>
                    
                    <channel type='unix'>
                      <source mode='bind'/>
                      <target type='virtio' name='org.qemu.guest_agent.0'/>
                      </channel>
                    
                    
                  </devices>
                    <cpu mode='host-model'>
                        <model fallback='allow'/><feature policy='require' name='svm'/></cpu>
                    
                    </domain>
karim created on local
```

```
☸ mytest in ~ 
➜ kcli --debug info vm karim 
<domain type='kvm' id='1'>
  <name>karim</name>
  <uuid>563a0bd4-cacb-44fc-a399-8a90c4825e7f</uuid>
  <metadata>
    <kvirt:info xmlns:kvirt="kvirt">
      <kvirt:creationdate>18-02-2020 08:14</kvirt:creationdate>
      <kvirt:profile>rhel8reg</kvirt:profile>
      <kvirt:image>rhel-8.1-x86_64-kvm.qcow2</kvirt:image>
      <kvirt:plan>kvirt</kvirt:plan>
    </kvirt:info>
  </metadata>
  <memory unit='KiB'>2097152</memory>
  <currentMemory unit='KiB'>2097152</currentMemory>
  <vcpu placement='static'>2</vcpu>
  <resource>
    <partition>/machine</partition>
  </resource>
  <os>
    <type arch='x86_64' machine='pc-i440fx-4.1'>hvm</type>
    <boot dev='hd'/>
    <bootmenu enable='yes'/>
  </os>
  <features>
    <acpi/>
    <apic/>
    <pae/>
  </features>
  <cpu mode='custom' match='exact' check='full'>
    <model fallback='forbid'>EPYC-IBPB</model>
    <vendor>AMD</vendor>
    <feature policy='require' name='x2apic'/>
    <feature policy='require' name='tsc-deadline'/>
    <feature policy='require' name='hypervisor'/>
    <feature policy='require' name='tsc_adjust'/>
    <feature policy='require' name='arch-capabilities'/>
    <feature policy='require' name='cmp_legacy'/>
    <feature policy='require' name='perfctr_core'/>
    <feature policy='require' name='virt-ssbd'/>
    <feature policy='require' name='rdctl-no'/>
    <feature policy='require' name='skip-l1dfl-vmentry'/>
    <feature policy='require' name='mds-no'/>
    <feature policy='disable' name='monitor'/>
    <feature policy='disable' name='ibpb'/>
    <feature policy='require' name='svm'/>
    <feature policy='require' name='topoext'/>
  </cpu>
  <clock offset='utc'/>
  <on_poweroff>destroy</on_poweroff>
  <on_reboot>restart</on_reboot>
  <on_crash>restart</on_crash>
  <devices>
    <emulator>/usr/bin/qemu-system-x86_64</emulator>
    <disk type='file' device='disk'>
      <driver name='qemu' type='qcow2'/>
      <source file='/var/lib/libvirt/images/first/karim_0.img'/>
      <backingStore type='file' index='1'>
        <format type='qcow2'/>
        <source file='/var/lib/libvirt/images/vm-base-images/rhel-8.1-x86_64-kvm.qcow2'/>
        <backingStore/>
      </backingStore>
      <target dev='vda' bus='virtio'/>
      <alias name='virtio-disk0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x05' function='0x0'/>
    </disk>
    <disk type='file' device='cdrom'>
      <driver name='qemu' type='raw'/>
      <source file='/var/lib/libvirt/images/first/karim.ISO'/>
      <backingStore/>
      <target dev='hdd' bus='ide'/>
      <readonly/>
      <alias name='ide0-1-1'/>
      <address type='drive' controller='0' bus='1' target='0' unit='1'/>
    </disk>
    <controller type='usb' index='0' model='piix3-uhci'>
      <alias name='usb'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x2'/>
    </controller>
    <controller type='pci' index='0' model='pci-root'>
      <alias name='pci.0'/>
    </controller>
    <controller type='ide' index='0'>
      <alias name='ide'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x1'/>
    </controller>
    <controller type='virtio-serial' index='0'>
      <alias name='virtio-serial0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x04' function='0x0'/>
    </controller>
    <interface type='network'>
      <mac address='52:54:00:de:2a:14'/>
      <source network='default' portid='4522bf24-41b8-4c8d-9494-9b90b25806fa' bridge='virbr0'/>
      <target dev='vnet0'/>
      <model type='virtio'/>
      <alias name='net0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
    </interface>
    <serial type='pty'>
      <source path='/dev/pts/1'/>
      <target type='isa-serial' port='0'>
        <model name='isa-serial'/>
      </target>
      <alias name='serial0'/>
    </serial>
    <console type='pty' tty='/dev/pts/1'>
      <source path='/dev/pts/1'/>
      <target type='serial' port='0'/>
      <alias name='serial0'/>
    </console>
    <channel type='unix'>
      <source mode='bind' path='/var/lib/libvirt/qemu/channel/target/domain-1-karim/org.qemu.guest_agent.0'/>
      <target type='virtio' name='org.qemu.guest_agent.0' state='connected'/>
      <alias name='channel0'/>
      <address type='virtio-serial' controller='0' bus='0' port='1'/>
    </channel>
    <input type='tablet' bus='usb'>
      <alias name='input0'/>
      <address type='usb' bus='0' port='1'/>
    </input>
    <input type='mouse' bus='ps2'>
      <alias name='input1'/>
    </input>
    <input type='keyboard' bus='ps2'>
      <alias name='input2'/>
    </input>
    <graphics type='spice' port='5900' autoport='yes' listen='127.0.0.1'>
      <listen type='address' address='127.0.0.1'/>
    </graphics>
    <video>
      <model type='cirrus' vram='16384' heads='1' primary='yes'/>
      <alias name='video0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x02' function='0x0'/>
    </video>
    <memballoon model='virtio'>
      <alias name='balloon0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x06' function='0x0'/>
    </memballoon>
  </devices>
  <seclabel type='dynamic' model='selinux' relabel='yes'>
    <label>system_u:system_r:svirt_t:s0:c60,c898</label>
    <imagelabel>system_u:object_r:svirt_image_t:s0:c60,c898</imagelabel>
  </seclabel>
  <seclabel type='dynamic' model='dac' relabel='yes'>
    <label>+107:+107</label>
    <imagelabel>+107:+107</imagelabel>
  </seclabel>
</domain>

name: karim
creationdate: 18-02-2020 08:14
status: up
autostart: False
image: rhel-8.1-x86_64-kvm.qcow2
user: cloud-user
plan: kvirt
profile: rhel8reg
cpus: 2
memory: 2048
net interface: eth0 mac: 52:54:00:de:2a:14 net: default type: routed
diskname: vda disksize: 20GB diskformat: file type: qcow2 path: /var/lib/libvirt/images/first/karim_0.img
```

```
☸ mytest in ~ 
➜ kcli --debug delete vm karim 
Are you sure? [y/N]: y
Deleting vm karim on local
Traceback (most recent call last):
  File "/usr/bin/kcli", line 11, in <module>
    load_entry_point('kcli==99.0', 'console_scripts', 'kcli')()
  File "/usr/lib/python3.7/site-packages/kvirt/cli.py", line 2552, in cli
    args.func(args)
  File "/usr/lib/python3.7/site-packages/kvirt/cli.py", line 187, in delete_vm
    result = k.delete(name, snapshots=snapshots)
  File "/usr/lib/python3.7/site-packages/kvirt/kvm/__init__.py", line 1587, in delete
    volume.delete(0)
  File "/usr/lib64/python3.7/site-packages/libvirt.py", line 3593, in delete
    if ret == -1: raise libvirtError ('virStorageVolDelete() failed', vol=self)
libvirt.libvirtError: cannot remove directory '/var/lib/libvirt/images/first': Device or resource busy
```

**Comment by wzzrd** [comment]
If it helps at all, here are my config.yml and profiles.yml:

https://pastebin.com/KQWZHieM
https://pastebin.com/sBY848L1

**Comment by karmab** [comment]
can i also ask a kcli list disk?

**Comment by wzzrd** [comment]
```
kcli --debug list disk
Listing disks...
+----------------------------------------------------------------------------+----------------+--------------------------------------------------------------------------------------------------------+
| Name                                                                       |      Pool      |                                                  Path                                                  |
+----------------------------------------------------------------------------+----------------+--------------------------------------------------------------------------------------------------------+
| 17763.379.190312-0539.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us.iso |      iso       | /var/lib/libvirt/images/iso/17763.379.190312-0539.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us.iso |
| CentOS-6-x86_64-GenericCloud.qcow2                                         | vm-base-images |               /var/lib/libvirt/images/vm-base-images/CentOS-6-x86_64-GenericCloud.qcow2                |
| CentOS-7-x86_64-DVD-1810.iso                                               |      iso       |                        /var/lib/libvirt/images/iso/CentOS-7-x86_64-DVD-1810.iso                        |
| CentOS-7-x86_64-GenericCloud.qcow2                                         | vm-base-images |               /var/lib/libvirt/images/vm-base-images/CentOS-7-x86_64-GenericCloud.qcow2                |
| CentOS-8-x86_64-1905-boot.iso                                              |      iso       |                       /var/lib/libvirt/images/iso/CentOS-8-x86_64-1905-boot.iso                        |
| CentOS-8-x86_64-Unofficial.qcow2                                           | vm-base-images |                /var/lib/libvirt/images/vm-base-images/CentOS-8-x86_64-Unofficial.qcow2                 |
| Fedora-Cloud-Base-31-1.9.x86_64.qcow2                                      | vm-base-images |              /var/lib/libvirt/images/vm-base-images/Fedora-Cloud-Base-31-1.9.x86_64.qcow2              |
| Fedora-Server-dvd-x86_64-29-1.2.iso                                        |      iso       |                    /var/lib/libvirt/images/iso/Fedora-Server-dvd-x86_64-29-1.2.iso                     |
| Fedora-Server-netinst-x86_64-30-1.2.iso                                    |      iso       |                  /var/lib/libvirt/images/iso/Fedora-Server-netinst-x86_64-30-1.2.iso                   |
| RHEL-8.0.0-20190326.n.0-x86_64-dvd1.iso                                    |      iso       |                  /var/lib/libvirt/images/iso/RHEL-8.0.0-20190326.n.0-x86_64-dvd1.iso                   |
| RHEL-8.1.0-20191015.0-x86_64-dvd1.iso                                      |      iso       |                   /var/lib/libvirt/images/iso/RHEL-8.1.0-20191015.0-x86_64-dvd1.iso                    |
| SLE-12-SP3-Server-DVD-x86_64-GM-DVD1.iso                                   |      iso       |                  /var/lib/libvirt/images/iso/SLE-12-SP3-Server-DVD-x86_64-GM-DVD1.iso                  |
| SLE-12-SP3-Server-DVD-x86_64-GM-DVD2.iso                                   |      iso       |                  /var/lib/libvirt/images/iso/SLE-12-SP3-Server-DVD-x86_64-GM-DVD2.iso                  |
| Win10_1703_EnglishInternational_x64.iso                                    |      iso       |                  /var/lib/libvirt/images/iso/Win10_1703_EnglishInternational_x64.iso                   |
| Win10_1709_EnglishInternational_x64.iso                                    |      iso       |                  /var/lib/libvirt/images/iso/Win10_1709_EnglishInternational_x64.iso                   |
| Windows7.iso                                                               |      iso       |                                /var/lib/libvirt/images/iso/Windows7.iso                                |
| Windows_Server_2016_Datacenter_EVAL_en-us_14393_refresh.iso                |      iso       |        /var/lib/libvirt/images/iso/Windows_Server_2016_Datacenter_EVAL_en-us_14393_refresh.iso         |
| __DIRECT_IO_TEST__                                                         |      iso       |                             /var/lib/libvirt/images/iso/__DIRECT_IO_TEST__                             |
| amq_0.img                                                                  |     first      |                                /var/lib/libvirt/images/first/amq_0.img                                 |
| arch-openstack-LATEST-image-bootstrap.qcow2                                | vm-base-images |           /var/lib/libvirt/images/vm-base-images/arch-openstack-LATEST-image-bootstrap.qcow2           |
| bootstrap.test.ocp4.lan.qcow2                                              |    default     |                       /var/lib/libvirt/images/ssd/bootstrap.test.ocp4.lan.qcow2                        |
| cirros-0.4.0-x86_64-disk.img                                               | vm-base-images |                  /var/lib/libvirt/images/vm-base-images/cirros-0.4.0-x86_64-disk.img                   |
| debian-10-openstack-amd64.qcow2                                            | vm-base-images |                 /var/lib/libvirt/images/vm-base-images/debian-10-openstack-amd64.qcow2                 |
| debian-9-openstack-amd64.qcow2                                             | vm-base-images |                 /var/lib/libvirt/images/vm-base-images/debian-9-openstack-amd64.qcow2                  |
| debian-9.7.0-amd64-DVD-1.iso                                               |      iso       |                        /var/lib/libvirt/images/iso/debian-9.7.0-amd64-DVD-1.iso                        |
| debian-9.8.0-amd64-netinst.iso                                             |      iso       |                       /var/lib/libvirt/images/iso/debian-9.8.0-amd64-netinst.iso                       |
| devmachine.deployment6.lan-disk1                                           |     first      |                     /var/lib/libvirt/images/first/devmachine.deployment6.lan-disk1                     |
| elmer-elgert.deployment6.lan-disk1                                         |     first      |                    /var/lib/libvirt/images/first/elmer-elgert.deployment6.lan-disk1                    |
| f29test.qcow2                                                              |     first      |                              /var/lib/libvirt/images/first/f29test.qcow2                               |
| f30template.qcow2                                                          |     first      |                            /var/lib/libvirt/images/first/f30template.qcow2                             |
| first                                                                      |  default_old   |                                     /var/lib/libvirt/images/first                                      |
| foreman-discovery-image-3.5.3-1.iso                                        |      iso       |                    /var/lib/libvirt/images/iso/foreman-discovery-image-3.5.3-1.iso                     |
| idm8.deployment6.lan-disk1                                                 |     first      |                        /var/lib/libvirt/images/first/idm8.deployment6.lan-disk1                        |
| ignition.qcow2                                                             |  default_old   |                                 /var/lib/libvirt/images/ignition.qcow2                                 |
| iso                                                                        |  default_old   |                                      /var/lib/libvirt/images/iso                                       |
| karim_0.img                                                                |     first      |                               /var/lib/libvirt/images/first/karim_0.img                                |
| master-01.dev.ocp4.lan.qcow2                                               |    default     |                        /var/lib/libvirt/images/ssd/master-01.dev.ocp4.lan.qcow2                        |
| master02.qcow2                                                             |     first      |                              /var/lib/libvirt/images/first/master02.qcow2                              |
| master03.qcow2                                                             |     second     |                             /var/lib/libvirt/images/second/master03.qcow2                              |
| openSUSE-Leap-15.2-OpenStack.x86_64.qcow2                                  | vm-base-images |            /var/lib/libvirt/images/vm-base-images/openSUSE-Leap-15.2-OpenStack.x86_64.qcow2            |
| openshift-nfs-exports.qcow2                                                |     first      |                       /var/lib/libvirt/images/first/openshift-nfs-exports.qcow2                        |
| registry-nfs.qcow2                                                         |     second     |                           /var/lib/libvirt/images/second/registry-nfs.qcow2                            |
| rhcos-4.1.0-x86_64-installer.iso                                           |      iso       |                      /var/lib/libvirt/images/iso/rhcos-4.1.0-x86_64-installer.iso                      |
| rhcos-4.2.0-x86_64-installer.iso                                           |      iso       |                      /var/lib/libvirt/images/iso/rhcos-4.2.0-x86_64-installer.iso                      |
| rhcos-4.2.0-x86_64-metal-bios.raw.gz                                       |      iso       |                    /var/lib/libvirt/images/iso/rhcos-4.2.0-x86_64-metal-bios.raw.gz                    |
| rhel-8.1-x86_64-kvm.qcow2                                                  | vm-base-images |                    /var/lib/libvirt/images/vm-base-images/rhel-8.1-x86_64-kvm.qcow2                    |
| rhel-guest-image-6.9-206.x86_64.qcow2                                      | vm-base-images |              /var/lib/libvirt/images/vm-base-images/rhel-guest-image-6.9-206.x86_64.qcow2              |
| rhel-server-6.10-update-11-x86_64-kvm.qcow2                                | vm-base-images |           /var/lib/libvirt/images/vm-base-images/rhel-server-6.10-update-11-x86_64-kvm.qcow2           |
| rhel-server-6.10-x86_64-dvd.iso                                            |      iso       |                      /var/lib/libvirt/images/iso/rhel-server-6.10-x86_64-dvd.iso                       |
| rhel-server-6.8-x86_64-dvd.iso                                             |      iso       |                       /var/lib/libvirt/images/iso/rhel-server-6.8-x86_64-dvd.iso                       |
| rhel-server-6.9-x86_64-dvd.iso                                             |      iso       |                       /var/lib/libvirt/images/iso/rhel-server-6.9-x86_64-dvd.iso                       |
| rhel-server-7.4-x86_64-dvd.iso                                             |      iso       |                       /var/lib/libvirt/images/iso/rhel-server-7.4-x86_64-dvd.iso                       |
| rhel-server-7.5-x86_64-dvd.iso                                             |      iso       |                       /var/lib/libvirt/images/iso/rhel-server-7.5-x86_64-dvd.iso                       |
| rhel-server-7.6-x86_64-dvd.iso                                             |      iso       |                       /var/lib/libvirt/images/iso/rhel-server-7.6-x86_64-dvd.iso                       |
| rhel-server-7.7-update-2-x86_64-kvm.qcow2                                  | vm-base-images |            /var/lib/libvirt/images/vm-base-images/rhel-server-7.7-update-2-x86_64-kvm.qcow2            |
| rhel6-template-20151103.qcow2                                              |     second     |                      /var/lib/libvirt/images/second/rhel6-template-20151103.qcow2                      |
| rhel7-template-20150523.qcow2                                              |     second     |                      /var/lib/libvirt/images/second/rhel7-template-20150523.qcow2                      |
| rhel7desktop.deployment6.lan-disk1                                         |     first      |                    /var/lib/libvirt/images/first/rhel7desktop.deployment6.lan-disk1                    |
| rhel8-demo-2.deployment6.lan-disk1                                         |     first      |                    /var/lib/libvirt/images/first/rhel8-demo-2.deployment6.lan-disk1                    |
| rhel8-demo.deployment6.lan-disk1                                           |     first      |                     /var/lib/libvirt/images/first/rhel8-demo.deployment6.lan-disk1                     |
| rhv-m.deployment6.lan-disk1                                                |     second     |                       /var/lib/libvirt/images/second/rhv-m.deployment6.lan-disk1                       |
| rhv-m.deployment6.lan-disk1bk                                              |     second     |                      /var/lib/libvirt/images/second/rhv-m.deployment6.lan-disk1bk                      |
| rhv-node-01.deployment6.lan-disk1                                          |     first      |                    /var/lib/libvirt/images/first/rhv-node-01.deployment6.lan-disk1                     |
| rhv-node-02.deployment6.lan-disk1                                          |     first      |                    /var/lib/libvirt/images/first/rhv-node-02.deployment6.lan-disk1                     |
| rhv-node-03.deployment6.lan                                                |     second     |                       /var/lib/libvirt/images/second/rhv-node-03.deployment6.lan                       |
| rhv-tools-setup.iso                                                        |      iso       |                            /var/lib/libvirt/images/iso/rhv-tools-setup.iso                             |
| sat6cast_backup_20190207.qcow2                                             |     second     |                     /var/lib/libvirt/images/second/sat6cast_backup_20190207.qcow2                      |
| sat6cast_new.qcow2                                                         |    default     |                             /var/lib/libvirt/images/ssd/sat6cast_new.qcow2                             |
| second                                                                     |  default_old   |                                     /var/lib/libvirt/images/second                                     |
| ssd                                                                        |  default_old   |                                      /var/lib/libvirt/images/ssd                                       |
| systemrescuecd-6.0.3.iso                                                   |     second     |                        /var/lib/libvirt/images/second/systemrescuecd-6.0.3.iso                         |
| tower.deployment6.lan-disk1                                                |    default     |                        /var/lib/libvirt/images/ssd/tower.deployment6.lan-disk1                         |
| tower8.deployment6.lan-disk1                                               |     first      |                       /var/lib/libvirt/images/first/tower8.deployment6.lan-disk1                       |
| ubuntu-19.10-server-cloudimg-amd64.img                                     | vm-base-images |             /var/lib/libvirt/images/vm-base-images/ubuntu-19.10-server-cloudimg-amd64.img              |
| virtio-win-1.9.6.iso                                                       |      iso       |                            /var/lib/libvirt/images/iso/virtio-win-1.9.6.iso                            |
| vm-base-images                                                             |  default_old   |                                 /var/lib/libvirt/images/vm-base-images                                 |
| wayne-unsworth.deployment6.lan-disk1                                       |     first      |                   /var/lib/libvirt/images/first/wayne-unsworth.deployment6.lan-disk1                   |
| win2k19.qcow2                                                              |     second     |                              /var/lib/libvirt/images/second/win2k19.qcow2                              |
| windows_7_home_premium_with_sp1_x64_dvd_u_676549.iso                       |      iso       |            /var/lib/libvirt/images/iso/windows_7_home_premium_with_sp1_x64_dvd_u_676549.iso            |
| windows_7_professional_with_sp1_x86_dvd_631025.iso                         |      iso       |             /var/lib/libvirt/images/iso/windows_7_professional_with_sp1_x86_dvd_631025.iso             |
| worker01.dev.ocp4.lan.qcow2                                                |    default     |                        /var/lib/libvirt/images/ssd/worker01.dev.ocp4.lan.qcow2                         |
| worker02.qcow2                                                             |     first      |                              /var/lib/libvirt/images/first/worker02.qcow2                              |
+----------------------------------------------------------------------------+----------------+--------------------------------------------------------------------------------------------------------+
```

```
sudo virsh pool-list
 Name             State    Autostart
--------------------------------------
 default          active   yes
 default_old      active   no
 first            active   yes
 iso              active   yes
 second           active   yes
 vm-base-images   active   yes
```

```
$ sudo virsh pool-dumpxml default
<pool type='dir'>
  <name>default</name>
  <uuid>ccaeb042-98c4-4834-8549-dfa47c9df3b6</uuid>
  <capacity unit='bytes'>536608768000</capacity>
  <allocation unit='bytes'>258915037184</allocation>
  <available unit='bytes'>277693730816</available>
  <source>
  </source>
  <target>
    <path>/var/lib/libvirt/images/ssd</path>
    <permissions>
      <mode>0755</mode>
      <owner>0</owner>
      <group>0</group>
      <label>system_u:object_r:virt_image_t:s0</label>
    </permissions>
  </target>
</pool>


$ sudo virsh pool-dumpxml first
<pool type='dir'>
  <name>first</name>
  <uuid>01a9ccce-717c-40f6-8ef8-e40e098b28f7</uuid>
  <capacity unit='bytes'>751357132800</capacity>
  <allocation unit='bytes'>173280763904</allocation>
  <available unit='bytes'>578076368896</available>
  <source>
  </source>
  <target>
    <path>/var/lib/libvirt/images/first</path>
    <permissions>
      <mode>0755</mode>
      <owner>0</owner>
      <group>0</group>
      <label>system_u:object_r:virt_image_t:s0</label>
    </permissions>
  </target>
</pool>


$ sudo virsh pool-dumpxml vm-base-images
<pool type='dir'>
  <name>vm-base-images</name>
  <uuid>14220362-8487-4ec0-8af8-7ce3905ca4f6</uuid>
  <capacity unit='bytes'>53660876800</capacity>
  <allocation unit='bytes'>10155163648</allocation>
  <available unit='bytes'>43505713152</available>
  <source>
  </source>
  <target>
    <path>/var/lib/libvirt/images/vm-base-images</path>
    <permissions>
      <mode>0775</mode>
      <owner>0</owner>
      <group>0</group>
      <label>system_u:object_r:unlabeled_t:s0</label>
    </permissions>
  </target>
</pool>
```

**Comment by karmab** [comment]
addressed in https://github.com/karmab/kcli/commit/bf6645cde27d6ae39d3b5ee4487ead74374ef123
